pkg/piecestore/rpc/server/psdb/psdb_test.go

```diff
@@ -0,0 +1,324 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package psdb
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	_ "github.com/mattn/go-sqlite3"
+	"github.com/stretchr/testify/assert"
+	pb "storj.io/storj/protos/piecestore"
+
+	"golang.org/x/net/context"
+)
+
+var ctx = context.Background()
+var parallelCount = 100
+
+func TestOpenPSDB(t *testing.T) {
+	tests := []struct {
+		it  string
+		err string
+	}{
+		{
+			it:  "should successfully create database",
+			err: "",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.it, func(t *testing.T) {
+			assert := assert.New(t)
+
+			tmp, err := ioutil.TempDir("", "example")
+			if err != nil {
+				log.Fatal(err)
+			}
+			defer os.RemoveAll(tmp)
+
+			dbpath := filepath.Join(tmp, "test.db")
+
+			DB, err := OpenPSDB(dbpath)
+			if tt.err != "" {
+				assert.NotNil(err)
+				assert.Equal(tt.err, err.Error())
+				return
+			}
+			assert.Nil(err)
+			assert.NotNil(DB)
+			assert.NotNil(DB.DB)
+		})
+	}
+}
+
+func TestDeleteTTLByID(t *testing.T) {
+	tests := []struct {
+		it  string
+		id  string
+		err string
+	}{
+		{
+			it:  "should successfully Delete TTL by ID",
+			id:  "butts",
+			err: "",
+		},
+	}
+
+	tmp, err := ioutil.TempDir("", "example")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer os.RemoveAll(tmp)
+
+	dbpath := filepath.Join(tmp, "test.db")
+	db, err := OpenPSDB(dbpath)
+	if err != nil {
+		t.Errorf("Failed to create database")
+		return
+	}
+	defer os.Remove(dbpath)
+
+	for _, tt := range tests {
+		for i := 0; i < parallelCount; i++ {
+			t.Run(tt.it, func(t *testing.T) {
+				t.Parallel()
+				assert := assert.New(t)
+
+				db.mtx.Lock()
+				db.DB.Exec(fmt.Sprintf(`INSERT or REPLACE INTO ttl (id, created, expires) VALUES ("%s", "%d", "%d")`, tt.id, time.Now().Unix(), 0))
```

Comment:
perhaps you could test this by actually inserting a file instead? this test is assuming quite a bit about the internals. if the internals change, even in a reasonable way, now we have to update the implied schema here too